jbpm-event-emitters/jbpm-event-emitters-elasticsearch/src/main/java/org/jbpm/event/emitters/elasticsearch/DefaultESInstanceViewTransformerFactory.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jbpm.event.emitters.elasticsearch;
+
+import java.text.SimpleDateFormat;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.fasterxml.jackson.databind.MapperFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import org.jbpm.persistence.api.integration.InstanceView;
+import org.jbpm.persistence.api.integration.model.CaseInstanceView;
+import org.jbpm.persistence.api.integration.model.ProcessInstanceView;
+import org.jbpm.persistence.api.integration.model.TaskInstanceView;
+import org.jbpm.persistence.api.integration.model.TaskOperationView;
+
+
+public class DefaultESInstanceViewTransformerFactory implements ESInstanceViewTransformerFactory{
+    
+    private final String dateFormatStr = System.getProperty("org.jbpm.event.emitters.elasticsearch.date_format", System
+            .getProperty("org.kie.server.json.date_format", "yyyy-MM-dd'T'HH:mm:ss.SSSZ"));
+    private final boolean ignoreNull = Boolean.getBoolean("org.jbpm.event.emitters.elasticsearch.ignoreNull");
+    
+    protected Map<Class<?>, ESInstanceViewTransformer> viewMapping = new ConcurrentHashMap<>();
+     
+    public DefaultESInstanceViewTransformerFactory() {
+        DefaultESInstanceViewTransformer taskView = new DefaultESInstanceViewTransformer("tasks", "task");
+        viewMapping.put(TaskOperationView.class, taskView);
+        viewMapping.put(TaskInstanceView.class, taskView);
+        viewMapping.put(ProcessInstanceView.class, new DefaultESInstanceViewTransformer("processes", "process"));
+        viewMapping.put(CaseInstanceView.class, new DefaultESInstanceViewTransformer("cases", "case"));
+    }
```

Comment:
I might missing the logic somewhere but just in case to double check:
The logic I cannot find is to load any other views the user might create for himself.
lookup for ESInstanceViewTransformer and set it class to itself. that way we allow users to do their own. Now that we created a proper framework for this we can allo the user to do it by himself

Reply:
the idea I have in mind is that users should provide (or extend this default implementation) and specify this system property `org.jbpm.event.emitters.elasticsearch.factory` to replace default
https://github.com/kiegroup/jbpm/pull/1932/files#diff-204ae3aefeb9d9ef9abd24afbfa15e08b53048301b8b220646008791630432d7R73-R83